### PR TITLE
feat(model): Add SuperADD VAND4.0 winning method

### PR DESCRIPTION
**Pull request by alexriedel1**
## 📝 Description
Hello!
This adds the SuperADD as proposed in https://arxiv.org/abs/2605.14808 using the authors implementation https://github.com/LukasRoom/SuperADD

I made some minor changes to improve the memory consumption and runtime and to allow for batching during validation. Also the default image is set to 448 as 640 patches on the ViT H Super takes up extremely much memory. 

I also removed all the "bells and whistles" like segmentation map post processing and training augmentations. The augmentations can easily be made with anomalibs datamodules and the segmentation map processing can be done manually if needed but only boost the score i guess. 

With the default settings the model reaches:

MVTec AD2 Vial
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.9995918273925781     │
│       image_F1Score       │    0.9756097793579102     │
│        pixel_AUROC        │    0.9501352906227112     │
│       pixel_F1Score       │     0.357709139585495     │
└───────────────────────────┴───────────────────────────┘
```
MVTec AD2 Wallplug
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.5274074077606201     │
│       image_F1Score       │    0.5432098507881165     │
│        pixel_AUROC        │    0.9193798303604126     │
│       pixel_F1Score       │    0.4700793921947479     │
└───────────────────────────┴───────────────────────────┘
```

## ✨ Changes

Select what type of change your PR is:

- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔄 Refactor (non-breaking change which refactors the code base)
- [ ] ⚡ Performance improvements
- [ ] 🎨 Style changes (code style/formatting)
- [ ] 🧪 Tests (adding/modifying tests)
- [ ] 📚 Documentation update
- [ ] 📦 Build system changes
- [ ] 🚧 CI/CD configuration
- [ ] 🔧 Chore (general maintenance)
- [ ] 🔒 Security update
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).
- [ ] 🏷️ My PR title follows conventional commit format.

For more information about code review checklists, see the [Code Review Checklist](https://github.com/open-edge-platform/anomalib/blob/main/docs/source/markdown/guides/developer/contributing.md).

### CODE REVIEW

File: src/anomalib/models/image/super_add/__init__.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2022-2025 Intel Corporation
```
Comment (samet-akcay) [inline src/anomalib/models/image/super_add/__init__.py]:
```suggestion
# Copyright (C) 2026 Intel Corporation
```

File: src/anomalib/models/image/super_add/__init__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+SuperADD is an anomaly detection model that builds a memory bank of patch
+features extracted from a pretrained DINOv3 Vision Transformer. Multi-layer ViT
+token features are computed over overlapping image patches, a per-layer memory
+bank is built from normal training images via distance-based coreset
+subsampling, and test images are scored by nearest-neighbor distance to the
+bank.
+
+Example:
+    >>> from anomalib.data import MVTecAD
```
Comment (samet-akcay) [inline src/anomalib/models/image/super_add/__init__.py]: would it be an idea to include `MVTecAD2` as the main class in the documentation? As it is the main dataset used in the paper ?

File: src/anomalib/models/image/super_add/lightning_model.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
+# Copyright (C) 2022-2025 Intel Corporation
```
Comment (samet-akcay) [inline src/anomalib/models/image/super_add/lightning_model.py]:
```suggestion
# Copyright (C) 2026 Intel Corporation
```

File: src/anomalib/models/image/super_add/torch_model.py [hunk 4]
```diff
@@ -0,0 +1,637 @@
+# Copyright (C) 2022-2025 Intel Corporation
```
Comment (samet-akcay) [inline src/anomalib/models/image/super_add/torch_model.py]:
```suggestion
# Copyright (C) 2026 Intel Corporation
```

File: src/anomalib/models/__init__.py [hunk 5]
```diff
@@ -135,7 +137,9 @@ class UnknownModelError(ModuleNotFoundError):
     "Patchflow",
     "ReverseDistillation",
     "Stfpm",
+    "SuperADD",
     "Supersimplenet",
+    "SuperADD",
```
Comment (samet-akcay) [inline src/anomalib/models/__init__.py]: do we need this ? 

```suggestion
```

**Comment by samet-akcay** [comment]
thanks @alexriedel1, great addition! 

Would it be an idea to document the following in the pr description: 
- results in the paper vs pr results in a table side-by-side? 

Also potentially documenting steps to do additionally to match the performance reported in the paper? 

thanks again!

**Comment by alexriedel1** [comment]
> thanks @alexriedel1, great addition!
> 
> Would it be an idea to document the following in the pr description:
> 
> * results in the paper vs pr results in a table side-by-side?
> 
> Also potentially documenting steps to do additionally to match the performance reported in the paper?
> 
> thanks again!

Yes! Im working on some docs and still finding a good default setting Trade Off between memory consumption and accuracy. The papers Input sizes will likely cause OOM errors for most Users.

File: src/anomalib/models/image/super_add/__init__.py (hunk 2 above)
Comment (alexriedel1) [reply src/anomalib/models/image/super_add/__init__.py]: yes i fixed it

**Comment by alexriedel1** [comment]
I added the docs including some more explanation how to get the paper results!

**Comment by alexriedel1** [comment]
I'm still fixing a bug. In the original paper, the anomaly score didn't matter and wasn't calculated. With using the max anomaly map values as anomaly score, we fail when there exist two many patches (large image size and small image patch sizes defined in the model) and no validation images (as in MVTecAD2). We need to use the mean top 10% highest values of the anomaly maps and have to blur the maps like i patchcore to make the anomaly score value work better.

**Comment by alexriedel1** [comment]
OK as MVTecAD2 has no abnormal Validation samples, anomalib F1 scores just take the highest prediction score during training as threshold. In the original paper, a set of train images is held out to calculate the treshold as the 95% percentile of 1/8th of the Training Images. The pixel threshold is x1.421 of the image threshold. This leads to much better results for MVTecAD 2 (and maybe for all datasets without validation data?). I implemented the custom pre-processor, but ist a worth a thought if this should be applied as default for datasets without val data. 

Now we get with only half the training data and much smaller resolution almost paper results (Paper Pixel F1 wallplugs 0.79)

```
Only 60% of training images wallplugs
Input Image Resize(1024)
SuperADD(patch_size=512, patch_overlap=128)

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.6187962889671326     │
│       image_F1Score       │    0.6404494643211365     │
│        pixel_AUROC        │    0.9446907043457031     │
│       pixel_F1Score       │    0.7019270062446594     │
└───────────────────────────┴───────────────────────────┘
```

File: src/anomalib/models/image/super_add/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+## Architecture
+
+![SuperADD Architecture](/docs/source/images/super_add/architecture.png "SuperADD Architecture")
+
+## Usage
+
+`anomalib train --model SuperAdd --data MVTecAD2 --data.category <category>`
+
+The original papers authors use a very high input resolution to the model. To replicate their input stratetgy, scale the MVTecAD2 images by a factor of 0.625 and keep the aspect ration of your images for better accuracy. Set the models paramters to `patch_size=640` and `patch_overlap=128`. These settings result in a high memory consumption.
```
Comment (ashwinvaidya17) [inline src/anomalib/models/image/super_add/README.md]:
```suggestion
The original papers authors use a very high input resolution to the model. To replicate their input stratetgy, scale the MVTecAD2 images by a factor of 0.625 and keep the aspect ratio of your images for better accuracy. Set the models paramters to `patch_size=640` and `patch_overlap=128`. These settings result in a high memory consumption.
```

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks for the efforts! Really clean implementation. I've created a branch for the next minor release `feature/v2.6.0`. Can you target it?

**Comment by alexriedel1** [comment]
> Thanks for the efforts! Really clean implementation. I've created a branch for the next minor release `feature/v2.6.0`. Can you target it?

Thanks!
Sure I did: https://github.com/open-edge-platform/anomalib/pull/3660

**Comment by ashwinvaidya17** [comment]
> > Thanks for the efforts! Really clean implementation. I've created a branch for the next minor release `feature/v2.6.0`. Can you target it?
> 
> Thanks! Sure I did: #3660

Sorry I meant targeting this branch to feature 2.6.0. The idea is that every model, dataset, or any another added feature will be shipped in a minor release.

**Comment by alexriedel1** [comment]
> > > Thanks for the efforts! Really clean implementation. I've created a branch for the next minor release `feature/v2.6.0`. Can you target it?
> > 
> > 
> > Thanks! Sure I did: #3660
> 
> Sorry I meant targeting this branch to feature 2.6.0. The idea is that every model, dataset, or any another added feature will be shipped in a minor release.

Ah Sorry I closed the other PR and changed the base branch here. The Integration Tests Are failing because of Draem but I dont know really why

**Review by ashwinvaidya17 — APPROVED** [review]

**Comment by ashwinvaidya17** [comment]
> > > > Thanks for the efforts! Really clean implementation. I've created a branch for the next minor release `feature/v2.6.0`. Can you target it?
> > > 
> > > 
> > > Thanks! Sure I did: #3660
> > 
> > 
> > Sorry I meant targeting this branch to feature 2.6.0. The idea is that every model, dataset, or any another added feature will be shipped in a minor release.
> 
> Ah Sorry I closed the other PR and changed the base branch here. The Integration Tests Are failing because of Draem but I dont know really why

Thanks. I'll sort out the tests
